src/tools/updatePackageDependencies.ts

```diff
@@ -186,14 +191,18 @@ export async function updatePackageDependencies(): Promise<void> {
     fs.writeFileSync('package.json', content);
 }
 
-function replaceVersion(fileName: string, newVersion: string): string;
-function replaceVersion(fileName: undefined, newVersion: string): undefined;
-function replaceVersion(fileName: string | undefined, newVersion: string): string | undefined;
-function replaceVersion(fileName: string | undefined, newVersion: string): string | undefined {
+function replaceVersion(fileName: string, oldVersion: string, newVersion: string): string;
+function replaceVersion(fileName: undefined, oldVersion: string, newVersion: string): undefined;
+function replaceVersion(fileName: string | undefined, oldVersion: string, newVersion: string): string | undefined;
+function replaceVersion(fileName: string | undefined, oldVersion: string, newVersion: string): string | undefined {
     if (fileName === undefined) {
         return undefined; // If the file name is undefined, no version to replace
     }
 
+    if (oldVersion.length > 0) {
+        return fileName.replaceAll(oldVersion, newVersion);
```

Comment:
The regex was removing the -net6.0 from O# install paths because it considered it a preview tag. Made sending in the old version string optional so that replacing it can be more precise.